AppKit/CPTextView/CPTextView.j

```diff
@@ -0,0 +1,1890 @@
+    _maxSize = CGSizeMake(_frame.size.width, 1e7);
+
+    [self setRichText:NO];
+    _usesFontPanel = YES;
+    _allowsUndo = YES;
+    _isVerticallyResizable = YES;
+    _isHorizontallyResizable = NO;
+
+    _caretRect = CGRectMake(0, 0, 1, 11);
+}
+
+
+#pragma mark -
+#pragma mark Copy and past methods
+
+- (void)copy:(id)sender
+{
+   _copySelectionGranularity = _previousSelectionGranularity;
+   [super copy:sender];
+}
+
+- (void)paste:(id)sender
+{
+    if (_copySelectionGranularity > 0)
+    {
+        if (![self _isCharacterAtIndex:MAX(0, _selectionRange.location - 1) granularity:_copySelectionGranularity])
+            [self insertText:" "];
+    }
+    [super paste:sender];
+
+    if (_copySelectionGranularity > 0)
+    {
+        if (![self _isCharacterAtIndex:CPMaxRange(_selectionRange) granularity:_copySelectionGranularity])
+            [self insertText:" "];
+    }
+}
+
+#pragma mark -
+#pragma mark Responders method
+
+- (BOOL)acceptsFirstResponder
+{
+    if ([self isSelectable])
+        return YES;
+
+    return NO;
+}
+
+- (BOOL)becomeFirstResponder
+{
+    _isFirstResponder = YES;
+    [self updateInsertionPointStateAndRestartTimer:YES];
+    [[CPFontManager sharedFontManager] setSelectedFont:[self font] isMultiple:NO];
+    [self setNeedsDisplay:YES];
+
+    return YES;
+}
+
+- (BOOL)resignFirstResponder
+{
+    [_caretTimer invalidate];
+    _caretTimer = nil;
+    _isFirstResponder = NO;
+    [self setNeedsDisplay:YES];
+
+    return YES;
+}
+
+
+#pragma mark -
+#pragma mark Delegate methods
+
+/*!
+    TODO : documentation
+*/
+- (void)setDelegate:(id <CPTextViewDelegate>)aDelegate
+{
+    if (aDelegate === _delegate)
+        return;
+
+    var notificationCenter = [CPNotificationCenter defaultCenter];
+
+    _delegateRespondsToSelectorMask = 0;
+
+    if (_delegate)
+    {
+        [notificationCenter removeObserver:_delegate name:CPTextDidChangeNotification object:self];
+        [notificationCenter removeObserver:_delegate name:CPTextViewDidChangeSelectionNotification object:self];
+        [notificationCenter removeObserver:_delegate name:CPTextViewDidChangeTypingAttributesNotification object:self];
+    }
+
+    _delegate = aDelegate;
+
+    if (_delegate)
+    {
+        if ([_delegate respondsToSelector:@selector(textDidChange:)])
+            [notificationCenter addObserver:_delegate selector:@selector(textDidChange:) name:CPTextDidChangeNotification object:self];
+
+        if ([_delegate respondsToSelector:@selector(textViewDidChangeSelection:)])
+            [notificationCenter addObserver:_delegate selector:@selector(textViewDidChangeSelection:) name:CPTextViewDidChangeSelectionNotification object:self];
+
+        if ([_delegate respondsToSelector:@selector(textViewDidChangeTypingAttributes:)])
+            [notificationCenter addObserver:_delegate selector:@selector(textViewDidChangeTypingAttributes:) name:CPTextViewDidChangeTypingAttributesNotification object:self];
```

Comment:
This is going to add memory leaks in the application. You should take a look to the commit 392a6cfb to see how to deal with this kind of delegate